app/controllers/posts_controller.bk.rb

```diff
@@ -0,0 +1,66 @@
+class PostsController < ApplicationController
+    before_action :find_post, only: [:show, :edit, :update, :destroy]
+    before_action :authenticate_user!, except: [:show, :index]
+
+
+    def index
+        @string = params[:search]
+        @posts = Post.order(created_at: :desc).search(@string).page(params[:page]).per(7)
```

Comment:
Another magic number. This per page number is different than the one I saw earlier